google/cloud/storage/internal/grpc_hmac_key_metadata_parser.cc

```diff
@@ -0,0 +1,67 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "google/cloud/storage/internal/grpc_hmac_key_metadata_parser.h"
+#include "google/cloud/internal/time_utils.h"
+
+namespace google {
+namespace cloud {
+namespace storage {
+GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_BEGIN
+namespace internal {
+
+HmacKeyMetadata GrpcHmacKeyMetadataParser::FromProto(
+    google::storage::v2::HmacKeyMetadata const& rhs) {
+  HmacKeyMetadata result;
+  result.id_ = rhs.id();
+  result.access_id_ = rhs.access_id();
+  // The protos use `projects/{project}` format, but the field may be absent or
+  // may have a project id (instead of number), we need to do some parsing. We
+  // are forgiving here. It is better to drop one field rather than dropping
+  // the full message.
+  if (rhs.project().rfind("projects/", 0) == 0) {
+    result.project_id_ = rhs.project().substr(std::strlen("projects/"));
```

Comment:
Perhaps we should factor out the `"projects/"`?

Or, perhaps you might care for something like ...
```
  absl::string_view project = rhs.project();
  absl::ConsumePrefix(&project, "projects/");
  result.project_id_ =  std::string(project);
```

Reply:
Done.